corehq/apps/reports/standard/users/reports.py

```diff
@@ -179,15 +177,39 @@ def _html_list(changes, unstyled=True):
 
     properties = UserHistoryReport.get_primary_properties(domain)
     properties.pop("user_data", None)
-    primary_changes = {
-        properties.get(key, key): value for key, value in changes.items()
-        if key in properties
-    }
-    more_count = len(changes) - len(primary_changes)
+    primary_changes = {}
+
+    for key, value in changes.items():
+        if key in properties:
+            if key == 'commcare_location_id':
+                primary_changes[_("primary location (mobile users only)")] = cached_location_id_to_name(value)
```

Comment:
Is this intended to be only for mobile workers or we had to do it because its only available for mobile workers.? It will be possible to do it for both web and mobile workers after https://github.com/dimagi/commcare-hq/pull/30253/files#diff-4284bb62f43ca4b0545a51692cec107399e2d26d955097a6893752c50efa4609R135 change is rolled out.

Reply:
It's mobile workers only because I thought this report only tracks changes to users themselves, not to related objects like DomainMembership. The linked PR looks like it's updating message structure - is it also starting to track DomainMembership changes?

Reply:
> only tracks changes to users themselves

and in addition to that we do track the changes like roles/locations/groups as well.

> The linked PR looks like it's updating message structure

Yes, its going to store messages in a more structured format. and also bring more consistency to how we track web and mobile users (to remove the confusion like we have here about locations). The tracking has not really been bound to what is exactly on the user only but about any update to the user including information about their permissions and access control or anything that the domain admin would be interested in knowing or something we would want to track.